manual/opt.tex

```diff
@@ -153,9 +153,110 @@ \subsubsection{quartic}
 \end{lstlisting}
 
 \subsubsection{adaptive}
-%To activate this optimizer, add ``-D BUILD\_LMYENGINE\_INTERFACE=1'' in the CMake command line.
 
-Manual is needed here.
+The default setting of the adaptive optimizer is to construct the linear method Hamiltonian and overlap matrices explicitly and add different shifts to the Hamiltonian matrix 
+as ``stabilizers''.
+The generalized eigenvalue problem is solved for each shift to obtain updates to the wave function parameters.
+Then a correlated sampling is performed for each shift's updated wave function and the initial trial wave function
+using the middle shift's updated wave function as the guiding function.
+The cost function for these wave functions is compared, and the update corresponding to the best cost function is selected.
+In the next iteration, the median magnitude of the stabilizers is set to that that generated the best update in the current iteration, thus adapting the magnitude of
+the stabilizers automatically.
+
+Thanks to recently developments in excited state variational principles, the adaptive optimizer is also able to optimize individual excited states directly.  
+In this case, it tries to minimize the following function: 
+\begin{equation*}
+\Omega[\Psi]=\frac{\left<\Psi|\omega-H|\Psi\right>}{\left<\Psi|{\left(\omega-H\right)}^2|\Psi\right>}
+\end{equation*}
+The global minimal of this functional corresponds to the state whose energy lies immediately above the shift parameter $\omega$ in the energy spectrum.
+For example, suppose the initial 
+trial wave function's parameters are taken as a guess for the first excited state.
+Then, by setting $\omega$ to be a value in between the ground state and first excited state 
+energies would cause the trial wave function to be optimized for the first excited state. 
```

Comment:
energy.  Now I see 'energies' refers to both 'ground state' and 'first excited state'
But it might be a little smoother as:
"Then, by setting $\omega$ to be an energy in between the ground state and the first excited state, the wave function will be optimized for the first excited state."

Another thought - the excited state optimization paragraph might be better moved to it's own subsection, or at least moved to end of the adaptive section.   The description is here because of shared implementation, but the idea of optimizing for excited states is logically separate.

Also - is omega only used for excited state optimization, or is it used for ground state optimization too?
If only for excited states, the bullet point about 'omega' in Additional information, and the recommendation should all be moved to follow the description paragraph.